docs/spec/index.md

```diff
@@ -139,3 +141,120 @@ Because some events may overlap in time, it may not be possible to choose a sing
+
+### `employee_rosters.txt`
+
+Describes which employees are scheduled to which rosters on which dates.
+
+Primary Key: `(roster_id,start_date)`
+
+| **Field Name** | **Type** | **Required** | **Description** |
+| --- | --- | --- | --- |
+| `employee_id` | ID | Required | |
+| `roster_id` | ID referencing `rosters.txt` | Required | |
+| `start_date` | Date | Required | |
+| `end_date` | Date | Required | |
+
+Each roster can only be assigned to one employee on each date. Employees may be scheduled to more than one roster on the same date.
+
+### `employee_run_dates.txt`
+
+Describes which employees are scheduled to which runs on which dates. If `employee_rosters.txt` is used, then describes exceptions to that schedule.
+
+Primary Key: `*`
+
+| **Field Name** | **Type** | **Required** | **Description** |
+| --- | --- | --- | --- |
+| `employee_id` | ID | Required | |
+| `date` | Date | Required | |
+| `service_id` | ID referencing `run_events.txt` | Required | Part of the Run ID, which is refered to as `(service_id, run_id)`. |
+| `run_id` | ID referencing `run_events.txt` | Required | The run that's either added or removed from this employee's schedule. If `exception_type` is `2` and `run_id` is not blank, then it must match a Run ID that the employee was scheduled to do on this date according to `employee_rosters.txt`, `rosters.txt` and `roster_dates.txt`. |
+| `exception_type` | Enum | Optional | `1` (or blank) - The run is assigned to this employee on the specified date.<br />`2` - The employee will not work this run on this date. |
+
+If a feed doesn't represent rosters, it can still assign employees to runs by putting every run for every date in this file. In that case, the `exception_type` column can be omitted because every row would be adding a date, which is the default when the column is blank.
+
+Each run can only be assigned to one employee on each date. Employees may be scheduled to more than one run on the same date.
+
+### `vehicle_assignments.txt`
+
+Primary Key: `*`
+
+| Field Name | Type | Required | Description |
+|---|---|---|---|
+| `date` | Date | Required |  |
+| `service_id` | ID referencing `calendar.service_id` or `calendar_dates.service_id` | Optional | Identifies a set of dates when the run is scheduled to take place. Required if `block_id`s are repeated between different `service_id`s. |
+| `block_id`		  | ID referencing `trips.block_id` | Conditionally required | Identifies the block. Either `trip_id` or `block_id` must be specified. |
+| `trip_id` | ID referencing `trips.trip_id` | Conditionally required | Either `trip_id` or `block_id` must be specified. In the case where both are supplied, `trip_id` overrides `block_id`. Note: multiple vehicles are allowed on the same block+date, but this is not recommended. |
```

Comment:
Would an agency ever schedule multiple vehicles to the same block? Doesn't that go against the definition of block? I know that might happen as a realtime adjustment, but if it's scheduled to happen, shouldn't they change the schedule so it's multiple different blocks?

If we can make that assumption, then this field could be removed.

Reply:
I included `trip_id` as an option for cases where `block_id` is not supplied in the GTFS. So, on 2nd thought I'd favor removing, unless we know this is a case to plan for out of the gate.

Reply:
f3281fe

Reply:
I'd recommend including it for specific trips. This works for agencies that don't publish blocks and may not wish to make specific trip-specific block assignments in TODS, and likewise works for agencies with interchangeable operations. (I could see us making these assignments on a trip-specific level to specify requirements by trip prior to building out cycles.)

Reply:
I'm in favour of keeping it as it is now, with the `trip_id` removed. 

If we allow assignments of individual trips to vehicles it breaks the logic of blocks as @skyqrose pointed out.

If we allow assignments of individual trips to vehicle _types_/_categories_ it overlaps content-wise with http://bit.ly/gtfs-vehicles (which already has a route-to-vehicle-category assignment modeling a "requirement"). The case that you pointed out last ("specify requirements by trip prior to building out cycles") should probably be modeled using http://bit.ly/gtfs-vehicles.

"agencies that don't publish blocks" can still wrap each trip in an artificial block and assign it this way and/or add their real blocks in the non-public `trips_supplement.txt` only. 

So overall, it seems like we can transmit the operational plan (including which vehicles/types will really be running the trips) without allowing to include `trip_id` in the assignments.